ContainerToolsForWindows/Public/BuildkitTools.psm1

```diff
@@ -146,130 +165,170 @@ function Initialize-BuildkitdService {
-                $command = "buildkitd.exe --register-service --debug --containerd-worker=true  --service-name buildkitd"
+                $command = "buildkitd.exe --register-service --debug --containerd-worker=true --service-name buildkitd"
             }
             Default {
-                Write-Warning "Failed to register buildkit service. Containerd conf file not found at $cniConfPath. Create the file to resolve this issue, then run this command $command"
-                return
+                Throw "Failed to register buildkit service. Containerd conf file not found at $cniConfPath. Create the file to resolve this issue, then run this command $command"
             }
-        } 
+        }
     }
-    
-    Invoke-Expression -Command $command
-    if ($LASTEXITCODE -gt 0) {
-        Write-Error "Failed to register buildkitd service."
+
+    $arguments = ($command -split " " | Select-Object -Skip 1) -join " "
+    $output = Invoke-ExecutableCommand -Executable $buildkitdExecutable -Arguments $arguments
+    if ($output.ExitCode -ne 0) {
+        Throw "Failed to register buildkitd service. $($output.StandardError.ReadToEnd())"
     }
 
-    sc.exe config buildkitd depend=containerd
-    if ($LASTEXITCODE -gt 0) {
-        Write-Error "Failed to set dependency for buildkitd on containerd."
+    $buildkitdService = Get-Service buildkitd -ErrorAction SilentlyContinue
+    if ($null -eq $buildkitdService ) {
+        Throw "Failed to register buildkitd service. $($Error[0].Exception.Message)"
     }
 
-    Get-Service *buildkitd* | Select-Object Name, DisplayName, ServiceName, ServiceType, StartupType, Status, RequiredServices, ServicesDependedOn
+    Set-Service buildkitd -StartupType Automatic
+    Write-Output "Successfully registered Buildkitd service.".
 
-    sc.exe query buildkitd
+    $output = Invoke-ExecutableCommand -Executable 'sc.exe' -arguments 'config buildkitd depend=containerd'
+    if ($output.ExitCode -ne 0) {
+        Write-Error "Failed to set dependency for buildkitd on containerd. $($output.StandardOutput.ReadToEnd())"
+    }
 
-    Write-Output "Successfully registered Buildkitd service."
-    Write-Information -InformationAction Continue -MessageData "To start buildkitd service, run 'Start-Service buildkitd' or 'Start-BuildkitdService'"
+    if ($Start) {
+        Start-BuildkitdService
+        Write-Output "Successfully started Buildkitd service."
+    }
+    else {
+        Write-Information -InformationAction Continue -MessageData "To start buildkitd service, run 'Start-Service buildkitd' or 'Start-BuildkitdService'"
+    }
 
     # YAGNI: Do we need a buildkitd.toml on Windows?
 }
 
 function Uninstall-Buildkit {
     param(
         [parameter(HelpMessage = "Buildkit path")]
-        [String] $Path
+        [String]$Path
     )
-    
-    Write-Output "Uninstalling buildkit"
-    if (!$path) {
-        $path = Get-DefaultInstallPath -Tool "Buildkit"
+
+    $tool = 'Buildkit'
```

Comment:
This function looks like it was meant to be a generic uninstall function, why isn’t it part of utilities with a tool param?

Reply:
This method calls several generic functions in the P[rivate module files](https://github.com/microsoft/containers-toolkit/tree/create-container-tools-module/ContainerToolsForWindows/Private) which take the paramenter `-Tool`. To avoid passing the word `Buildkit` in all the function, the `$tool = 'Buildkit'` is used.